timelock-api/src/main/conjure/cross-client-batched-timelock-api.yml

```diff
@@ -0,0 +1,63 @@
+types:
+  imports:
+    LeaderTime:
+      base-type: any
+      external:
+        java: com.palantir.lock.v2.LeaderTime
+    Long:
+      base-type: any
+      external:
+        java: java.lang.Long
+    LockWatchStateUpdate:
+      base-type: any
+      external:
+        java: com.palantir.lock.watch.LockWatchStateUpdate
+    ConjureIdentifiedVersion:
+      base-type: any
+      external:
+        java: com.palantir.atlasdb.timelock.api.ConjureIdentifiedVersion
+    GetCommitTimestampsRequest:
+      base-type: any
+      external:
+        java: com.palantir.atlasdb.timelock.api.GetCommitTimestampsRequest
+    GetCommitTimestampsResponse:
+      base-type: any
+      external:
+        java: com.palantir.atlasdb.timelock.api.GetCommitTimestampsResponse
+  definitions:
+    default-package: com.palantir.atlasdb.timelock.api
+    objects:
+      NamespacedLeaderTime:
+        fields:
+          namespace: string
+          leaderTime: LeaderTime
+      NamespacedGetCommitTimestampsRequest:
+        fields:
+          namespace: string
+          getCommitTimestampsRequest: GetCommitTimestampsRequest
+      NamespacedGetCommitTimestampsResponse:
+        fields:
+          namespace: string
+          getCommitTimestampsResponse: GetCommitTimestampsResponse
+
+services:
+  CrossClientBatchedConjureTimelockService:
+    name: Cross Client Batched Timelock Service
+    default-auth: header
+    package: com.palantir.atlasdb.timelock.batch.api
+    base-path: /tl/batch
+    endpoints:
+      leaderTimes:
+        http: POST /lts
+        args:
+          namespaces: set<string>
+        returns: list<NamespacedLeaderTime>
+        docs: |
+          Cross client batched version of ConjureTimelockService#leaderTime endpoint for acquiring leaderTimes.
+      getCommitTimestamps:
+        http: POST /gcts
+        args:
+          requests: list<NamespacedGetCommitTimestampsRequest>
```

Comment:
This one on the other hand is a bit messier - there's some nontrivial logic in how to reduce all of the legacy `GetCommitTimestampRequest`s into a single one (basically you can take the oldest easily, maybe the newest if there's some clever hackery on the client side).